docs/upgrade.rst

```diff
@@ -25,6 +25,13 @@ Usage of the ``PYBIND11_OVERLOAD*`` macros and ``get_overload`` function should
 be replaced by ``PYBIND11_OVERRIDE*`` and ``get_override``. In the future, the
 old macros may be deprecated and removed.
 
+The ``pybind11`` package on PyPI no longer fills the wheel "headers" slot - if
+you were using the headers from this slot, they are available by requesting the
+``global`` extra, that is, ``pip install "pybind11[global]"``. (Most users will
```

Comment:
Really stupid question: does this mean that there will be two separate packages on PyPI, or is ``[global]`` some kind of flag that affects the behavior of Pip when installing the package?

Reply:
Yes. The thing in brackets is called an "extra", short for extra requirements, and is supported everywhere in the Python packaging ecosystem; requirements lists, pip, pyproject.toml, etc. They are commonly used to add `[test]` requirements, for example - see [Pandas](https://github.com/pandas-dev/pandas/blob/6929e262dd22ac35baabf87a5236d451255fb66d/setup.py#L772-L779). You'll also see `setuptools_scm[toml]`, which is how you ensure you have the toml parser so you can use `pyproject.toml` + `setuptools_scm`. It can only enable extra packages as install requirements, not arbitrary changes. The extra requirement it enables in this case is a package called `pybind11-global`, which I'll explain above. But this is the public suggestion for adding it.